src/lib/registration/registrationInitialisation.ts

```diff
@@ -42,17 +42,17 @@ export const initBeacon = (): IRegistration => {
     emergencyContact3TelephoneNumber: "",
     emergencyContact3AlternativeTelephoneNumber: "",
 
-    uses: [initBeaconUse()],
+    uses: [initBeaconUse(true)],
   };
 };
 
 /**
  * Convenience function for returning an `empty` instance of a beacon use.
  *
- * @returns {BeaconUse}   JSON representation of a beacon use
+ * @param mainUse {boolean}     Whether the use is the main use for a beacon; defaults to false
```

Comment:
We currently treat the use at `useArray[0]` as the `main` use, `useArray[1]` as the `second` and so on.  Why the boolean flag?

Thinking about what the choice of having a `mainUse` flag means for, e.g. when user deletes their `mainUse` prior to submitting the registration.  Which of the remaining uses then becomes `main`?  Or if a user gets to check-your-answers and decides to re-order their uses.  I'm seeing an increase in complexity by introduction of the boolean.

Keen to understand your thinking :-)

Reply:
I think we discussed this on the API side of things.  We mentioned allowing the consumer to specify which was the main use for their beacon.

We could ignore the main use as the user submits the form and then just set it before we send it to the API instead? I think we do need to store it within the API, otherwise I don't think we could determine what was the users main use for their beacon?

Reply:
I remember the conversation now!  Sorry.  We decided:

> Relying on the ordering of a `uses` array to signify the ranking of beacon uses might be brittle if the array is re-ordered for whatever reason, whether by a developer unaware of the significance of the ordering or a bug.  It's also less expressive than explicitly declaring the individual use's priority.

Up to speed now.  What do you think about `usePriority: Integer` instead of `mainUse: boolean`? Then we also keep the semantic meaning of the second use, third use etc.  (I.e. there is a list of uses in priority order, instead of one `mainUse` use and then the rest).  

This could be done prior to sending to the API, as you say?